Functional requirements

Aderfish opened the pull request:
This PR adds the Functional requirements part to the PRD.

I have also added all pdfs and `__pycache__` to the `.gitignore`.

gruvw commented on template/7-functional_requirements.md:
```diff
@@ -1,22 +1,21 @@
 # Functional Requirements
 
-*Max 3 pages.*
+1. **User Authentication**: Users must be able to authenticate into their account to user the app. For this, we use pre-existing infrastructure, such as google authentication.
```
"such as Google Sign-in or other common OAuth providers".

gruvw commented on template/7-functional_requirements.md:
```diff
@@ -1,22 +1,21 @@
+1. **User Authentication**: Users must be able to authenticate into their account to user the app. For this, we use pre-existing infrastructure, such as google authentication.
 
-*List the key features of the MVP precisely.*
+2. **Post Creation**: Users must be able to create posts containing text. These posts will have embedded location data.
```
What about other forms of media like pictures and gifs. I think they are important to have for the MVP.

gruvw commented on template/7-functional_requirements.md:
```diff
@@ -1,22 +1,21 @@
 # Functional Requirements
 
```
This section is missing an introductory paragraph.

gruvw commented on template/7-functional_requirements.md:
```diff
@@ -1,22 +1,21 @@
-- Centauri Points
-- Challenges
-- Map Integration
+3. **Location-Based Feed**: Users must be able to see in a feed all of the posts that were made inside of a 100m radius around their current position. This feed will have different sorting options.
```
We need to describe the sorting options here.

gruvw commented on template/7-functional_requirements.md:
```diff
@@ -1,22 +1,21 @@
-- Map Integration
+3. **Location-Based Feed**: Users must be able to see in a feed all of the posts that were made inside of a 100m radius around their current position. This feed will have different sorting options.
 
+4. **Commenting System**: Users must be able to leave comments on posts. 
```
and see most recent comments first by default or most popular first if they prefer

gruvw commented on template/7-functional_requirements.md:
```diff
@@ -1,22 +1,21 @@
-*Describe key internal functionality.*
+6. **Voting System**: Users should be able to upvote or downvote posts and comments.
 
+7. **Posts and comments management**: Users should always be able see all of their own posts and comments, and delete them if they want to. Deleting a post automatically deletes all the comments and any information related to it.
```
We should mention the profile page explicitly here, or in a new separate item.

gruvw commented on template/7-functional_requirements.md:
```diff
@@ -1,22 +1,21 @@
 
+4. **Commenting System**: Users must be able to leave comments on posts. 
 
-*Include appropriate architectural diagrams.*
```
"include appropriate architectural diagrams." was removed but I don't see where it is included ?

Aderfish replied:
I added a paragraph in 23ac8bd184ad42e53ad384def3ced5d0403b87c2, but it feels a bit forced. Do you think there is one feature in particular for which it is relevant to explain the internal implementation ?

gruvw commented on template/7-functional_requirements.md:
```diff
@@ -1,22 +1,21 @@
+
+9. **Challenges**: Users get 3 challenges everyday. These are randomly picked from posts in their area. To complete these challenges, they have to travel to the post and open it.
+
+10. **Map Integration**: Users should be able to see all the nearby posts, all of their own posts and all their challenges on a map. It should be simple to navigate back and forth from the map to the other pages.
```
This is missing the seamless Google maps integration to see the travel route to any given pin.

gruvw reviewed (changes requested):
Great start on this chapter of the PRD. I left a few comments to fix.

Additionally, I feel like most of the items in the list are missing a reason behind it and that should be added here.

Example: Users should be able to upvote or downvote posts and comments. That way, the other users are able to quickly discard posts that received a negative opinion from the community, and it also allows for challenges to be generated on interesting posts.

Aderfish commented:
Thank you for your comments, I added more justifications for every feature and added the parts I was missing. I am still not sure how an architecture diagram is relevant here, or if we should do one for one feature in particular.

Aderfish commented on template/7-functional_requirements.md:
```diff
@@ -1,22 +1,29 @@
-*Max 3 pages.*
 
-*List the key features of the MVP precisely.*
+1. **User Authentication**: Users must be able to authenticate into their account to user the app. For this, we will use pre-existing infrastructure, such as Google Sign-in or other common OAuth providers.
```
fixed in cf3433e39a16593d0a217fc99d75b661c62ea097

Aderfish commented on template/7-functional_requirements.md:
```diff
@@ -1,22 +1,29 @@
-- Map Integration
+2. **Post Creation**: Users must be able to create posts. These can contain simple text or media, sucha as GIFs, photos or videos. These posts will have embedded location data. This will allow the users to share their experiences.
 
+3. **Location-Based Feed**: Users must be able to see in a feed all of the posts that were made inside of a 100m radius around their current position. This feed will have different sorting options. One option will show the most trending post first, wich will use a combination of recency and number of upvotes. Another will simply show the most upvoted posts first. Another wil show the closest post ...
```
done in cf3433e39a16593d0a217fc99d75b661c62ea097

Aderfish commented on template/7-functional_requirements.md:
```diff
@@ -1,22 +1,29 @@
-- Map Integration
+2. **Post Creation**: Users must be able to create posts. These can contain simple text or media, sucha as GIFs, photos or videos. These posts will have embedded location data. This will allow the users to share their experiences.
 
+3. **Location-Based Feed**: Users must be able to see in a feed all of the posts that were made inside of a 100m radius around their current position. This feed will have different sorting options. One option will show the most trending post first, wich will use a combination of recency and number of upvotes. Another will simply show the most upvoted posts first. Another wil show the closest post ...
```
done in 194a7c23baff2a8ac76421eb9d1ae44dad2fb596

Aderfish commented:
I addressed your comments and moved the imaged to the appendix as it was creating a page with only the image in the middle of other parts. Tell me what you think.

gruvw commented on template/7-functional_requirements.md:
```diff
@@ -1,22 +1,44 @@
+Users should be able to see all the nearby posts, all of their own posts and all their challenges on a map. It should be simple to navigate back and forth from the map to the other pages. It should also be possible to seamlessly open the Google maps app to navigate to any post. This allows the user to precisely see where each post is, and also to know where to go to complete challenges. 
+
+
+### Internal Architecture
```
only use two `##` here as it is not really a functional requirement (like the rest of the list) but more like another thing describing how everything will fit together.

Aderfish replied:
done in 30a5e59e9d56d1c5c32203ae59df6e769f1c549a

gruvw reviewed (changes requested):
Ok everything looks good to me now thank you so much for applying my suggestions.
